portfolio/src/main/java/com/google/sps/data/User.java

```diff
@@ -0,0 +1,26 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+public class User {
+  public String username;
+  public String comment;
```

Comment:
In general Java fields should be private. If you need to access these from outside the class you can add a public "getter" method to the class which returns the field value. (eg: public String getUserName() { return this.usrname;} ). 

This touches on a broader topic of encapsulation in object oriented programming. Which is definitely something we'll talk more about as you get into more complex server code, and I'm happy to answer questions about whenever.

TL;DR in this case is that, if these fields are public then a client of this class (eg: DataServlet.java) can directly modify the fields. For a simple data class like User that doesn't matter too much. But in a lot of other situations it can cause problems, so the convention is to always make fields private.

Reply:
Another alternative (instead of adding getters) here would be to make User an inner class of DataServlet.java. Encapsulation is not as pressing a concern for inner classes, so in that case it would be ok to leave the fields as 'public'.
Docs on inner classes (also called nested classes): https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html.